WordPress/Classes/Services/DomainsService.swift

```diff
@@ -28,12 +28,27 @@ struct DomainsService {
         remote.getDomainSuggestions(base: base,
                                     domainSuggestionType: domainSuggestionType,
                                     success: { suggestions in
-            success(suggestions)
+            let filteredSuggestions = self.filterSuggestions(suggestions, forBase: base)
+            success(filteredSuggestions)
         }) { error in
             failure(error)
         }
     }
 
+    // If any of the suggestions matches the base exactly,
+    // then filter that suggestion up to the top of the list.
+    fileprivate func filterSuggestions(_ suggestions: [DomainSuggestion], forBase base: String) -> [DomainSuggestion] {
```

Comment:
`filteredSuggestions` to me implies that you're _removing_ something unwanted here, a'la `.filter { }` method. how about `sortedSuggestions` or `rearrangedSuggestions`?

Reply:
Really good point – I'll tweak it :)